### ISSUE TITLE
New Rule: no-empty-statements

### ISSUE THREAD

**Issue by Fraegle** [palantir/tslint#648]
When having a ; after a block, like

``` JavaScript
    if (true) {
        // do something
    };
```

The current error is "statements are not aligned". While this is correct, it needs a little thinking, especially if the block is bigger and the end can be confused with an assignment like

``` JavaScript
    var myFunction = function() {
        // do something
    };
```

It would be nice, if the issue would be detected as empty statement and the ; pointed out as being unnecessary.

**Comment by JKillian** [comment]
Seems like a good idea!

**Comment by ajafff** [comment]
Recent versions of `semicolon` check for unnecessary empty statements. It doesn't matter if you specify the option `"always"` or `"never"`
